docs/architecture.md

```diff
@@ -0,0 +1,111 @@
+# Architecture
+
+This document describes the high-level architecture of nearcore.
+The focus here is on the implementation of the blockchain protocol, not the protocol itself.
+For reference documentation of the protocol, please refer to [nomicon](https://nomicon.io/)
+
+## Bird's Eye View
+
+![](images/architecture.svg)
+
+Since nearcore is an implementation of NEAR blockchain protocol, its goal is to produce a binary that runs as a blockchain client.
+More specifically, the neard binary can do the following:
+- generate files necessary to start a node
+- start a node with a given folder that contains required data
+
+## Entry Points
+
+`neard/src/main.rs` contains the main function that starts a blockchain node.
+However, this file mostly only contains the logic to parse arguments and dispatch different commands.
+
+`start_with_config` in `neard/src/lib.rs` is the actual entry point and it starts all the actors.
```

Comment:
@miraclx Once we decouple `neard`, we need to remember to update this description.